src/components/section/form-section.component.tsx

```diff
@@ -13,13 +13,45 @@ import { FormField, FormFieldProps, previousValue, SubmissionHandler } from '../
 import { FormContext } from '../../form-context';
 import { Tooltip } from '../inputs/tooltip/tooltip.component';
 import styles from './form-section.scss';
+import { evaluateExpression } from '../../utils/expression-runner';
+import dayjs from 'dayjs';
+import { parseToLocalDateTime } from '../../utils/form-helper';
 
 interface FieldComponentMap {
   fieldComponent: React.ComponentType<FormFieldProps>;
   fieldDescriptor: FormField;
   handler: SubmissionHandler;
 }
 
+const historicalValueTransformer = (field, obs) => {
```

Comment:
I believe we support this elsewhere in the framework. With existing support, what you're trying to achieve here has been abstracted within the [handlers](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/types.ts#L34). @arodidev Can we do away with the duplication?

Reply:
Would you suggest modifying the handler to accomodate instances where the obs is already present?

Reply:
@samuelmale I'm requesting some guidance on this before progressing, thanks.